src/main/java/seedu/address/model/pair/UniquePairList.java

```diff
@@ -0,0 +1,137 @@
+package seedu.address.model.pair;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.pair.exceptions.DuplicatePairException;
+import seedu.address.model.pair.exceptions.PairNotFoundException;
+
+/**
+ * A list of pairs that enforces uniqueness between its elements and does not allow nulls.
+ * A pair is considered unique by comparing using {@code Pair#isSamePair(Pair)}. As such, adding and updating of
+ * pairs uses Pair#isSamePair(Pair) for equality so as to ensure that the pair being added or updated is
+ * unique in terms of identity in the UniquePairList. However, the removal of a person uses Pair#equals(Object) so
+ * as to ensure that the pair with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Pair#isSamePair(Pair)
+ */
+public class UniquePairList implements Iterable<Pair> {
+
+    private final ObservableList<Pair> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Pair> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
```

Comment:
honestly, i don't think we should resort to using the current list implementation. if possible, can you make use of hashmaps instead? if not, we can KIV this, and create a corresponding issue for the future

Reply:
We will keep this in view